Sources/Common/UserDefaultsCache.swift

```diff
@@ -35,24 +42,18 @@ public class UserDefaultsCache<ObjectType: Codable> {
         let expires: Date
         let object: ObjectType
     }
-
-    private var subscriptionAppGroup: String?
+    
+    private var userDefaults: UserDefaults
     public private(set) var settings: UserDefaultsCacheSettings
-    private lazy var userDefaults: UserDefaults? = {
-        if let appGroup = subscriptionAppGroup {
-            return UserDefaults(suiteName: appGroup)
-        } else {
-            return UserDefaults.standard
-        }
-    }()
 
-    private let key: UserDefaultsCacheKey
+    private let key: UserDefaultsCacheKeyStore
 
-    public init(subscriptionAppGroup: String?, key: UserDefaultsCacheKey,
+    public init(userDefaults: UserDefaults = UserDefaults.standard,
```

Comment:
Actually we got rid of default param as part of https://github.com/duckduckgo/BrowserServicesKit/pull/755
For fragile parts of code (and here deciding if we use shared or not-shared user defaults) makes more sense to be explicit in the call instead of hiding it behind default value.

Reply:
I have removed subscriptionAppGroup because this class (UserDefaultsCache) is no longer part of subscription and adding that init would imply it's coupled. This is now an abstract class, and makes more sense to inject a fully configured UserDefaults object (or let it default to Standard)